modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuBarSkin.java

```diff
@@ -389,62 +276,141 @@ public MenuBarSkin(final MenuBar control) {
+                                boolean isRTL = control.getEffectiveNodeOrientation() == NodeOrientation.RIGHT_TO_LEFT;
+                                if (control.getScene().getWindow().isFocused()) {
+                                    if (openMenu != null && !openMenu.isShowing()) {
+                                        if (isRTL) {
+                                            moveToMenu(Direction.PREVIOUS, false); // just move the selection bar
+                                        } else {
+                                            moveToMenu(Direction.NEXT, false); // just move the selection bar
+                                        }
+                                        ev.consume();
+                                        return;
+                                    }
+                                    if (isRTL) {
+                                        moveToMenu(Direction.PREVIOUS, true);
+                                    } else {
+                                        moveToMenu(Direction.NEXT, true);
+                                    }
+                                }
+                                ev.consume();
+                                break;
+                            }
+                            case DOWN:
+                            //case SPACE:
+                            //case ENTER:
+                                // RT-18859: Doing nothing for space and enter
+                                if (control.getScene().getWindow().isFocused()) {
+                                    if (focusedMenuIndex != -1) {
+                                        Menu menuToOpen = getSkinnable().getMenus().get(focusedMenuIndex);
+                                        showMenu(menuToOpen, true);
+                                        ev.consume();
+                                    }
+                                }
+                                break;
+                            case ESCAPE:
+                                unSelectMenus();
+                                ev.consume();
+                                break;
+                        default:
+                            break;
+                        }
+                    }
+                });
+
+                // When we click else where in the scene - menu selection should be cleared.
+                sceneListenerHelper.addEventFilter(scene, MouseEvent.MOUSE_CLICKED, (ev) -> {
+                    Bounds containerScreenBounds = container.localToScreen(container.getLayoutBounds());
+                    if ((containerScreenBounds == null) || !containerScreenBounds.contains(ev.getScreenX(), ev.getScreenY())) {
+                        unSelectMenus();
+                    }
+                });
+
+                // When the parent window looses focus - menu selection should be cleared
+                sceneListenerHelper.addChangeListener(scene.windowProperty(), true, (sr, oldw, w) -> {
```

Comment:
```suggestion
                sceneListenerHelper.addChangeListener(scene.windowProperty(), true, w -> {
```

Reply:
my version does not create extra object(s).

Reply:
Hm, I thought that the number of objects created is the same, either a `ChangeListener` is created and then wrapped in `ChLi` or a `Consumer` is created and wrapped in `ChLi`.